ui/src/pages/frontsite/PatronProfile/PatronCurrentLoans/components/ExtendButton/ExtendButton.js

```diff
@@ -0,0 +1,112 @@
+import { Button, Icon, Popup } from 'semantic-ui-react';
+import { invenioConfig, ES_DELAY } from '@config';
+import _get from 'lodash/get';
+import _has from 'lodash/has';
+
+const expireDays = invenioConfig.circulation.loanWillExpireDays;
+
+export default class ExtendButton extends Component {
+  static INFO_MESSAGES = {
+    extendUrl: 'It is not possible to extend this loan!',
+    extendAction: reqDate =>
+      `It is too early for extending the loan. You can request an extension from ${reqDate}`,
+    maxExtensions:
+      'You have reached the max number of extensions for this loan!',
+    documentOverbook:
+      'There is a high demand for this literature, therefore is not possible to automatically extend the loan!',
+  };
+
+  handleExtendRequest = async () => {
+    const { loan, onExtendSuccess } = this.props;
+    const { document, patron_pid: patronPid } = loan.metadata;
+    const extendUrl = _get(loan, 'availableActions.extend');
+
+    await this.props.extendLoan(extendUrl, document.pid, patronPid);
+    setTimeout(() => {
+      onExtendSuccess();
+    }, ES_DELAY);
+  };
+
+  validateExtendUrl() {
+    return _has(this.props.loan, 'availableActions.extend');
+  }
+
+  validateExtendAction() {
+    return (
+      _get(this.props.loan, 'metadata.end_date').diffNow('days').days <=
+      expireDays
+    );
+  }
+
+  validateMaxExtensions() {
+    return (
+      _get(this.props.loan, 'metadata.extension_count', 0) <=
+      invenioConfig.circulation.extensionsMaxCount
+    );
+  }
+
+  validateDocumentOverbook() {
+    const isOverbooked = _get(
+      this.props.loan,
+      'metadata.document.circulation.overbooked',
+      false
+    );
+    return !isOverbooked;
+  }
+
+  validate = () =>
+    this.validateExtendUrl() &&
+    this.validateExtendAction() &&
+    this.validateMaxExtensions() &&
+    this.validateDocumentOverbook();
+
+  infoMessage = () => {
+    if (!this.validateExtendUrl())
+      return _get(ExtendButton.INFO_MESSAGES, 'extendUrl');
+    if (!this.validateExtendAction()) {
+      const extendAction = _get(ExtendButton.INFO_MESSAGES, 'extendAction');
+      const startReqDate = _get(this.props.loan, 'metadata.end_date')
+        .minus({ days: expireDays })
+        .toLocaleString({
+          month: 'long',
+          day: 'numeric',
+        });
+      return extendAction(startReqDate);
+    }
+    if (!this.validateMaxExtensions())
+      return _get(ExtendButton.INFO_MESSAGES, 'maxExtensions');
+    if (!this.validateDocumentOverbook())
+      return _get(ExtendButton.INFO_MESSAGES, 'documentOverbook');
+  };
+
+  render() {
+    const isDisabled = !this.validate();
+
+    return (
+      <>
+        <Button
+          color="purple"
+          size="mini"
+          content="Request extension"
+          disabled={isDisabled}
+          onClick={this.handleExtendRequest}
+        />
+        {isDisabled && (
+          <Popup
+            content={this.infoMessage()}
```

Comment:
here we evaluate 2 times the conditions. Not really a big issue, I am just wondering if there is a better solution, e.g. (not super elegant)
```
let result = true, msg = '';
!this.validateExtendUrl() ?
    result = false;
    msg = ExtendButton.INFO_MESSAGES['extendUrl'];
: !this.validateExtendAction() ?
    result = false;
    msg = ...
: ...
return {
    result: result,
    msg: msg
}
```